test/project_types/script/layers/infrastructure/script_service_test.rb

```diff
@@ -351,9 +213,56 @@
     end
   end
 
-  private
+  describe ".generate_module_upload_url" do
+    let(:user_errors) { [] }
+    let(:url) { nil }
+    let(:response) do
+      {
+        "data" => {
+          "moduleUploadUrlGenerate" => {
+            "url" => url,
+            "userErrors" => user_errors,
+          },
+        },
+      }
+    end
+
+    before do
+      api_client.expects(:query).returns(response)
```

Comment:
Should we continue to `stub` this instead of `expects`?  There is no side effects on the `api_client.query` that we care to ensure to have happened.  In this case, a stub is better instead of a mock.  WDYT?

Reply:
I think I tried originally but stubbing wasn't working (I'm probably doing something wrong). Looks fine now.